skimage/color/colorconv.py

```diff
@@ -804,6 +804,65 @@ def rgb2grey(rgb):
     return rgb2gray(rgb)
 
 
+def gray2rgba(image, alpha=None):
+    """Create a RGBA representation of a gray-level image.
+
+    Parameters
+    ----------
+    image : array_like
+        Input image.
+    alpha : array_like, optional
+        Alpha channel of the output image. It may be a scalar or an
+        array with the same shape as ``image``. If not specified it is
+        set to the maximum limit correspondin to ``image`` data type.
+
+    Returns
+    -------
+    rgba : ndarray
+        RGBA image. A new dimension of length 4 is added to input
+        image shape.
+
+    Raises
+    ------
+    ValueError
+        If ``alpha`` is a scala and is not in ``image`` data type
+        limits.
+    ValueError
+        If ``alpha`` is an array and have a different shape then
+        ``image``.
+    ValueError
+        If ``alpha`` is an array and have a different data type then
+        ``image``.
+
+    """
+
+    arr = np.asarray(image)
+
+    alpha_min, alpha_max = dtype_limits(arr, clip_negative=False)
+
+    if alpha is None:
+        alpha = alpha_max
+    elif np.isscalar(alpha):
+        if alpha_min > alpha or alpha > alpha_max:
+            raise ValueError("alpha is not in image data type limits.")
+        alpha_cast = arr.dtype.type(alpha)
+        if alpha_cast != alpha:
```

Comment:
The problem here is that you can't check for equality in floating point numbers.

```
In [13]: a = 0.3                                                                

In [14]: np.float32(a) == np.float64(a)                                         
Out[14]: False

In [15]: np.float32(a) == np.float64(np.float32(a))                             
Out[15]: True

In [16]: np.float64(np.float32(a)) == np.float64(a)                             
Out[16]: False
```

Reply:
You are right, the right way to do that is to use `np.can_cast`